### PR TITLE
Fix word breaks

### PR DESCRIPTION
**Pull request by KaichenWang**
**Why are these changes introduced?**

Mitigate visual issues caused by headings overflowing containers and word breaks

**What approach did you take?**

### 1. Remove usage of `hyphens` CSS property
Some fonts (example: Americana) do not properly support `hyphens: auto`, resulting in unnecessary breaks in words with no actual hyphen rendered.

### 2. Add `word-break: break-word;` for all headings
This prevents headings (which are typically set to larger font-sizes) from overflowing containers.

### 3. Change collection card headings to `H3` style instead of `H1` style

**Other considerations**

**Demo links**

- [Store](https://os2-demo.myshopify.com/?preview_theme_id=127499010070)
- [Editor](https://os2-demo.myshopify.com/admin/themes/127499010070/editor)

**Checklist**
- [ ] Followed [theme code principles](https://github.com/Shopify/dawn/blob/main/.github/CONTRIBUTING.md#theme-code-principles)
- [ ] Linted with [Theme Check](https://github.com/Shopify/theme-check)
- [ ] Tested on [mobile](https://shopify.dev/themes/store/requirements#mobile-browser-requirements)
- [ ] Tested on [multiple browsers](https://shopify.dev/themes/store/requirements#desktop-browser-requirements)
- [ ] Tested for [accessibility](https://shopify.dev/themes/best-practices/accessibility)

### CODE REVIEW

**Comment by KaichenWang** [comment]
Just want to note that @melissaperreault noticed that buttons (example: rich text) can also have issues with longer labels:

22-01-ir55g-13tmq

**Review by tyleralsbury — APPROVED** [review]
Tested headings for all sections, including content based ones in blocks like Image Banner, etc...

Also tested all main pages - product, collection, blog, article, etc...

Everything works well with regards to headings.  Far as I can tell, we are good.

**Review by melissaperreault — APPROVED** [review]
Thanks you Kai!!

Everything looks good except the Collage edge-case findings and other areas we could benefit from `word-break` as a quick follow up. I tested all instances a label could have a very long word.

**Edge-case**
- [ ] [Collage's Collection block](https://screenshot.click/22-01-j2esn-wybu0.png) going off screen. The parent container `card__information` [is stretching](https://screenshot.click/22-01-8yuxf-0n1jm.png), it seems.

**Other text findings**
- [ ] Rich text button [off screen](https://screenshot.click/22-01-ir55g-13tmq.png) 
- [ ] Featured Collection button [off screen](https://screenshot.click/22-01-1nf1z-k6fl3.png) 
- [ ] Blog posts [View all link](https://screenshot.click/22-01-exi7x-bqce0.png) 
- [ ] Blog comment [off screen](https://screenshot.click/22-01-cxi21-jp17x.png)
- [ ] [View full details on Featured product](https://screenshot.click/22-01-9qy6b-z0m6g.png) layout not optimal
- [ ] Image with text button [off screen](https://screenshot.click/22-01-r6dpk-eyhcy.png)
- [ ] Filter label on Collection template [would push the product count](https://screenshot.click/22-01-1w4bg-pth1n.png)
- [ ] Cart `Continue shopping` button [off screen](https://screenshot.click/22-01-gcgey-wamft.png)
- [ ] Cart notification: [All CTAs go off screen ](https://screenshot.click/22-01-rnu7b-jp61k.png)
- [ ] Cart template Continue shopping link [off screen](https://screenshot.click/22-01-fy7e4-l0jdw.png)
- [ ] Cart template Checkout button [off screen](https://screenshot.click/22-01-v1m35-0g6rz.png)
- [ ] Login: Forgot password long label [should remain left aligned ](https://screenshot.click/22-01-mh65b-47jkw.png)
- [ ] Login Sign in button [off screen](https://screenshot.click/22-01-ofjt5-dbb47.png)
- [ ] Account details: Long name or long address go [off screen](https://screenshot.click/22-01-ft91v-l76lj.png)
- [ ] Addresses: Return to Account and Add new address go [off screen](https://screenshot.click/22-01-s3x4q-wik2r.png) + to check the Address and details and the other buttons (Edit and Delete) going [off screen](https://screenshot.click/22-01-ozchf-it90l.png)

File: assets/component-image-with-text.css [hunk 1]
```diff
@@ -311,13 +311,11 @@
 
 .image-with-text__heading {
   word-break: break-word;
-  hyphens: auto;
```
Comment (LucasLacerdaUX) [inline assets/component-image-with-text.css:314]: > Remove usage of hyphens CSS property

Can you share a bit more context on the decision to remove hyphens? I probably missed this conversation, but would love to know the rationale behind this so we can have it documented :D

**Review by LucasLacerdaUX — APPROVED** [review]

**Comment by KaichenWang** [comment]
@melissaperreault Created follow up issue for items you noted. https://github.com/Shopify/dawn/issues/1273

**Pull request merged**